testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -260,6 +276,30 @@ public void testViewColumnAccessControl()
                 "Cannot select from columns \\[.*] in table .*.orders.*",
                 privilege(getSession().getUser(), "orders", SELECT_COLUMN));
 
+        // verify that role permissions are taken for the view owner's role grants
+        // even if executed by the owner with different roles enabled
+        executeExclusively(() -> {
+            systemSecurityMetadata.grantRoles(getSession(), Set.of("deniedrole"), Set.of(new TrinoPrincipal(USER, "test_view_access_owner")), false, Optional.empty());
+            try {
+                getQueryRunner().getAccessControl().denyIdentityTable((identity, table) -> !identity.getEnabledRoles().contains("deniedrole") || !"orders".equals(table));
```

Comment:
I would implement this differently without using deny. I view owner could have two roles where one has access to base table, then view owner with session when that role is not set would try to query the view and we would need to see it is passing while previously it was failing.

Reply:
I would like to do it that way too, but `TestingAccessControlManager` can only deny stuff. Unless there is a system access control installed where we can inject policies...